Fixed compatible TRs reading incorrectly.

TalonSabre opened the pull request:
TR compatibility for each pokemon had been saving to the same list as TM compatibility, and for some reason it was not working as intended. Seemed like an issue with the way the data merged, so I had TRs read and write to an entirely separate variable. After finishing, and seeing that it built and ran, I tested the impacted features, and confirmed it is working.

TalonSabre commented:
If I do anything incorrectly or backwards on GitHub, sorry. This is the first time I have really submitted anything to GitHub for use. This was also my first time using C#, so if there is a better way to write any of my code, I am open to learning.
This issue annoyed me enough that I taught myself the basics of C#: [SwSh TR wrong offsets.](https://github.com/kwsch/pkNX/issues/317)
I will probably attempt to fix the issue with evolutions next for the same reason.
Everyone who contributed to pkNX, thanks for the awesome work up to this point. Have a nice day!

duckdoom4 commented on pkNX.Structures/Personal/Interfaces/IMovesInfo.cs:
You should also update the SetIMoves function

TalonSabre replied:
I think I see why you say that now; it has to do with separating TRs and making IMovesInfo_SWSH its exclusive interface.

- I added an if to SetIMovesInfo, it wasn't an issue, assuming I did it right.
- I assume SetPersonalInfo in [IPersonalInfo](pkNX.Structures/Personal/Interfaces/IPersonalInfo.cs) needs a case for SWSH, since it otherwise appears to never use the if statement I added in SetIMovesInfo. IPersonalInfoSWSH does not inherit/use IPersonalInfo_1 or _2, so I might put this (with proper spacing and tabs):
`if (self is IPersonalInfoSWSH self_SWSH && other is IPersonalInfoSWSH other_SWSH)
{ self_SWSH.SetIMovesInfo(other_SWSH); }`
- SetPersonalInfo doesn't appear to ever call with two of the same IPersonalInfo interfaces, so a lot of the Set functions used in it can't set all the data. I am guessing it is intentional because each game has different data in those areas. Like SWSH has TMs, but the other games don't have the same ones, so it can only be set by other SWSH data.

I am going to go ahead and add that if to SetPersonalInfo. Hopefully that will be correct, otherwise I can just revert or fix the change.

duckdoom4 commented on pkNX.Structures/Dumpers/PersonalDumper.cs:
```diff
@@ -5,6 +5,7 @@
 
 namespace pkNX.Structures;
 
+/* Now unused
```
If it's unused, just delete it. That's what we have version control for

TalonSabre replied:
Thanks!
Originally I downloaded an old version of project without the evolutions issue to use, but it still had this bug. I actually started editing that, so I basically mimicked the changes I made in the old one.
I will look at the function (I may have just forgot to copy my edits, or else I didn't think about its use).
I will also delete the old function I commented out. You make a valid point. I should have already done that, and probably just mentioned it in my post. Now that I think about it, GitHub would display that I deleted it anyways.
Regardless, I will get to correcting things. Appreciate you taking a look and catching that stuff!

duckdoom4 reviewed (changes requested):
Looks good to me 👍🏼 (besides the small comments)
Nicely done for your first time using things!

TalonSabre commented:
> (besides the small comments)
Realizing I type a lot, I put the important bits in bullet points. If you have the time, you could read/skim over the rest.
- **For my future reference, would it be better to have longer/more specific comments or no comments for projects on this platform?**
I remember being told that keeping documentation is a good idea in general, and adding comments is one way to do that. My assumption is if I don't write comments, I would put the info somewhere like a commit description.
That said, some of the comments are probably redundant anyways, and I only put them in for consistency. For example, where the program iterates through TMs, it has "// TMs" next to it, so I added "// TRs" next to code I added. I guess I probably shouldn't bother with comments unless I know a way to describe the code when isn't self-explanatory?
I'm asking because, even though I don't know how much experience you have coding, you probably know what you're talking about. I'm relatively young, and I'm trying to learn the proper way to do things so that I can find work as a programmer. Part of the issue is that I've never had a teacher/professor say they use comments that much, and I've never been required to add any. As a result:
- **I tend not to comment, especially since it's difficult for me to keep things concise while being descriptive. That said, I did try to explain some logic with comments.**
- **Consider this the last thing I deliberately bother you specifically with. Thanks again, I genuinely appreciate you taking the time!**

duckdoom4 commented:
> I guess I probably shouldn't bother with comments unless I know a way to describe the code when isn't self-explanatory?

That's how I usually comment code. I try to think like this:
 - if I come back to this code in half a year, would I still understand it without comments?
- maybe I would understand, but would the average programmer
- If I work in an open source project, should I tell why I did something? Or maybe refer to some other logic so the next person who comes across this code knows where to look.

TalonSabre commented:
It has been a couple days since, but I am just now seeing this. I cannot thank you enough for all the feedback.
Your first three bullets make a lot of sense. The first one is something that I try to keep in mind at all times. The next two are things that I understand, but don't think about as much. I never thought about the fourth point before, so that is something I will keep in mind as well.

> (When I mentioned 'small comments', I was referring to my own small issues I pointed out about your commit :p)

It is still something I struggle with, so I went ahead and asked, but I didn't think that was what you meant xd

duckdoom4 commented:
Let get this merged 🎉
